onnxruntime/test/perftest/performance_runner.cc

```diff
@@ -62,10 +62,36 @@ Status PerformanceRunner::RunOneIteration(bool isWarmup) {
   return Status::OK();
 }
 
-PerformanceRunner::~PerformanceRunner() = default;
+static TestModelInfo* CreateModelInfo(const PerformanceTestConfig& performance_test_config_) {
+  if (CompareCString(performance_test_config_.backend.c_str(), ORT_TSTR("ort")) == 0) {
+    return TestModelInfo::LoadOnnxModel(performance_test_config_.model_info.model_file_path.c_str());
+  }
+  if (CompareCString(performance_test_config_.backend.c_str(), ORT_TSTR("tf")) == 0) {
+    return TFModelInfo::Create(performance_test_config_.model_info.model_file_path.c_str());
+  }
+  ORT_NOT_IMPLEMENTED(ToMBString(performance_test_config_.backend), " is not supported");
+}
 
+static TestSession* CreateSession(OrtEnv* env, const PerformanceTestConfig& performance_test_config_,
+                                  TestModelInfo* test_model_info) {
+  if (CompareCString(performance_test_config_.backend.c_str(), ORT_TSTR("ort")) == 0) {
+    return new OnnxRuntimeTestSession(env, performance_test_config_, test_model_info);
```

Comment:
Can we use make_unique/unique_ptr? I see a lot of places in the test code where we rely on manual new/deletes. This had come up in the PR that fixes the x86 test cases as well.

Reply:
I changed all of them into smart pointers, except the others don't need be freed at here.
Smart pointers are good, but they have performance cost, and they make the code hard to understand. One example is, the shared_ptr in Tensor class, is something we really want to remove.

Reply:
Also, there is no memory leak in onnx_test_runner. The x86 test failure is not caused by any memory leak.

Reply:
unique_ptr has no perf cost. Returning raw pointers from such factory functions is bound to cause issues if the caller is not careful. When you return a smart ptr you don't need to worry so much about how the caller is going to use/save the returned ptr.

Did you get a chance to verify the x86 failures Randy had reported?

Reply:
I've verified. InferenceSession get freed when RunSingleTestCase is returned. So, I don't get the point, why it is related to memory leak or smart pointers. There isn't any memory leak in onnx_test_runner.  I don't get the value of using smart pointers more. I don't see any benefit from doing that.

Reply:
I didn't mean to say that there was a memory leak in onnx_test_runner. What I meant was that when I was reviewing the x86 PR, I noticed similar things where new/delete were used manually. I am just making a suggestion here that if the caller of CreateSession is not careful with the handling of the returned ptr, it can lead to leaks and this kind of pattern causes others to follow suit. If a unique_ptr is returned, there is less chance of leaks; obviously I don't have to explain this.  Since this is test code, I'm less concerned.

Reply:
Back to the original discussion, these two functions are file local, I don't expect any there place would use them, except the constructor.

Reply:
And obviously, whenever you got a pointer from a function named CreateXXX, you get the ownership of the pointer and you need to free it. That's what 'Create' means.

Reply:
I would rather depend on the compiler to release resources than depending on humans/conventions, but since this is restricted to a local file and in test code, I'm less concerned.